src/Components/Patient/PatientPrivacyToggle.tsx

```diff
@@ -0,0 +1,153 @@
+        );
+        if (
+          res &&
+          res.status === 200 &&
+          res?.data &&
+          (res.data?.privacy == true || res.data?.privacy == false)
+        ) {
+          setPrivacy(res.data.privacy);
+        } else {
+          Notification.Error({
+            msg: "Failed to fetch privacy",
+          });
+        }
+      } catch (e) {
+        Notification.Error({
+          msg: "Something went wrong..!",
+        });
+      }
+    };
+    if (
+      consultation &&
+      consultationId &&
+      consultation?.current_bed?.id &&
+      consultation?.current_bed?.bed_object?.id
+    ) {
+      getPrivacyInfo();
+    }
+  }, [consultation]);
+
+  //function to toggle the privacy of the patient
+  const togglePrivacy = async () => {
+    try {
+      if (consultation?.current_bed?.id) {
+        const res = await dispatch(
+          togglePatientPrivacy(consultation?.current_bed?.id as string)
+        );
+        if (res && res.status === 200) {
+          setPrivacy(!privacy);
+          Notification.Success({
+            msg: "Privacy updated successfully",
+          });
+          if (fetchPatientData) fetchPatientData({ aborted: false });
+        } else if (res && res.status === 403) {
+          Notification.Error({
+            msg: res.data.detail,
+          });
+        } else {
+          Notification.Error({
+            msg: "Failed to update privacy",
+          });
+        }
+      }
+    } catch (e) {
+      Notification.Error({
+        msg: "Something went wrong..!",
+      });
+    }
+  };
+  if (allowPrivacyToggle() && consultation?.current_bed?.id)
+    return (
+      <div className="flex flex-row justify-start gap-2 pt-2">
+        <div className="tooltip rounded-md bg-gray-300  px-3 py-2 text-sm font-semibold">
+          Privacy Mode: {privacy ? "ON" : "OFF"}
+          <span className="tooltip-text tooltip-bottom -translate-x-1/2 text-sm">
+            privacy setting for camera feed visual
+          </span>
+        </div>
+        {!privacy ? (
+          <button
+            className=" tooltip items-center rounded-md bg-gray-300 p-1 text-red-500 hover:bg-red-500 hover:text-gray-200"
+            onClick={togglePrivacy}
+            id="privacy-toggle"
+          >
+            <CareIcon className="care-l-lock text-3xl" />
+            <span className="tooltip-text tooltip-bottom -translate-x-1/2 text-sm">
+              Lock Privacy
+            </span>
+          </button>
+        ) : (
+          <button
+            className="tooltip items-center rounded-md bg-gray-300 p-1 text-black hover:bg-gray-500 hover:text-gray-200"
+            onClick={togglePrivacy}
+            id="privacy-toggle"
+          >
+            <CareIcon className="care-l-unlock text-3xl" />
+            <span className="tooltip-text tooltip-bottom -translate-x-1/2 text-sm">
+              Unlock Privacy
+            </span>
+          </button>
+        )}
+      </div>
+    );
+
+  return <></>;
```

Comment:
negate the condition and revert the returns, will be easy to read

Reply:
done